python/semantic_kernel/utils/tracing/decorators.py

```diff
@@ -0,0 +1,233 @@
+    COMPLETION_EVENT_COMPLETION,
+    COMPLETION_TOKEN,
+    ERROR_TYPE,
+    FINISH_REASON,
+    MAX_TOKEN,
+    MODEL,
+    OPERATION,
+    PROMPT_EVENT,
+    PROMPT_EVENT_PROMPT,
+    PROMPT_TOKEN,
+    RESPONSE_ID,
+    SYSTEM,
+    TEMPERATURE,
+    TOP_P,
+)
+
+OTEL_ENABLED_ENV_VAR = "SEMANTICKERNEL_EXPERIMENTAL_GENAI_ENABLE_OTEL_DIAGNOSTICS"
+OTEL_SENSITIVE_ENABLED_ENV_VAR = "SEMANTICKERNEL_EXPERIMENTAL_GENAI_ENABLE_OTEL_DIAGNOSTICS_SENSITIVE"
+
+
+_enable_diagnostics = os.getenv(OTEL_ENABLED_ENV_VAR, "false").lower() in ("true", "1", "t")
+_enable_sensitive_events = os.getenv(OTEL_SENSITIVE_ENABLED_ENV_VAR, "false").lower() in ("true", "1", "t")
+
+# Creates a tracer from the global tracer provider
+tracer = trace.get_tracer(__name__)
+
+
+def are_model_diagnostics_enabled() -> bool:
+    """Check if model diagnostics are enabled.
+
+    Model diagnostics are enabled if either EnableModelDiagnostics or EnableSensitiveEvents is set.
+    """
+    return _enable_diagnostics or _enable_sensitive_events
+
+
+def are_sensitive_events_enabled() -> bool:
+    """Check if sensitive events are enabled.
+
+    Sensitive events are enabled if EnableSensitiveEvents is set.
+    """
+    return _enable_sensitive_events
+
+
+def trace_chat_completion(model_provider: str) -> Callable:
+    """Decorator to trace chat completion activities."""
+
+    def inner_trace_chat_completion(completion_func: Callable) -> Callable:
+        @functools.wraps(completion_func)
+        async def wrapper_decorator(*args: Any, **kwargs: Any) -> list[ChatMessageContent]:
+            chat_history: ChatHistory = kwargs["chat_history"]
+            settings: PromptExecutionSettings = kwargs["settings"]
+
+            if hasattr(settings, "ai_model_id") and settings.ai_model_id:
+                model_name = settings.ai_model_id
+            elif hasattr(args[0], "ai_model_id") and args[0].ai_model_id:
+                model_name = args[0].ai_model_id
+            else:
+                model_name = "unknown"
+
+            span = _start_completion_activity(model_name, model_provider, chat_history, settings)
+
+            try:
+                completions: list[ChatMessageContent] = await completion_func(*args, **kwargs)
+            except Exception as exception:
+                if span:
+                    _set_completion_error(span, exception)
+                    span.end()
+                raise
+
+            if span:
+                with trace.use_span(span, end_on_exit=True):
+                    if completions:
+                        first_completion = completions[0]
+                        response_id = first_completion.metadata.get("id", None)
+                        if not response_id:
+                            response_id = (
+                                first_completion.inner_content.get("id", None)
+                                if first_completion.inner_content
+                                else None
+                            )
```

Comment:
```suggestion
                        response_id = first_completion.metadata.get("id", None) or (
                                first_completion.inner_content.get("id", None)
                                if first_completion.inner_content
                                else None
                            )
```

Reply:
Simplified to 
response_id = first_completion.metadata.get("id") or (first_completion.inner_content or {}).get("id")